backend/devices/bitbox02/handlers/handlers.go

```diff
@@ -289,10 +289,17 @@ func (handlers *Handlers) postSetMnemonicPassphraseEnabled(r *http.Request) inte
 func (handlers *Handlers) getVersionHandler(_ *http.Request) interface{} {
 	currentVersion := handlers.device.Version()
 	newVersion := bitbox02bootloader.BundledFirmwareVersion(handlers.device.Product())
-
+	var newVersionStr string
+	if newVersion != nil {
+		newVersionStr = newVersion.String()
+	}
+	canUpgrade := false
+	if newVersion != nil {
+		canUpgrade = newVersion.AtLeast(currentVersion) && currentVersion.String() != newVersion.String()
+	}
```

Comment:
Can we just move `canUpgrade := false` (nit: I think for consistency `var canUpgrade bool` would be nicer) above and have everything in a single if?

```go
var newVersionStr string
var canUpgrade bool

if newVersion != nil {
                neeVersionStr = newVersion.String()
		canUpgrade = newVersion.AtLeast(currentVersion) && currentVersion.String() != newVersionStr
	}
```

(also: wdyt about changing the implementation of `String()` in `bitbox02-api-go` to add 
```go
if version == nil { return "" }
```
This way we wouldn't need a `newVersionStr`. But this might be out of scope for this PR :)

Reply:
Done the first part. I don't think changing `String()` is a good idea, as I think normally methods, even on pointer receivers, assume the instance is not nil, so returning `""` there is quite implicit. Explicit > impllicit. Looking at the call site one in that case would look like it would panic.